tests/deprecated_api/test_remove_1-8.py

```diff
@@ -271,6 +272,22 @@ def test_v1_8_0_deprecated_training_type_plugin_property():
         trainer.training_type_plugin
 
 
+def test_v1_8_0_deprecate_trainer_data_loading_mixin():
+    trainer = Trainer(max_epochs=1)
+    model = BoringModel()
+    dm = BoringDataModule()
+    trainer.fit(model, datamodule=dm)
+
+    with pytest.deprecated_call(
+        match=r"`TrainerDataLoadingMixin.prepare_dataloader` was deprecated in v1.6 and will be removed in v1.8.",
+    ):
+        trainer.prepare_dataloader(dataloader=model.train_dataloader, shuffle=False)
+    with pytest.deprecated_call(
+        match=r"`TrainerDataLoadingMixin.request_dataloader` was deprecated in v1.6 and will be removed in v1.8.",
+    ):
+        trainer.request_dataloader(stage=RunningStage.TRAINING)
```

Comment:
offtopic comment: when this is removed we can get rid of the optional `model=...` argument from call_hook

Reply:
Yes! looking forward to it :)